solutions/ai-chatgpt/components/ChatBrowser.tsx

```diff
@@ -0,0 +1,137 @@
+import { useEffect, useState } from 'react'
+import { useCookies } from 'react-cookie'
+
+import { type Message, ChatLine, LoadingChatLine } from './ChatLine'
+import { InputMessage } from './InputMessage'
+import InputKey from './InputKey'
+import { generatePromptFromMessages } from '../lib/generatePromptFromMessages'
+
+const COOKIE_NAME = 'nextjs-example-ai-chat-gpt3-user'
+const COOKIE_KEY = 'nextjs-example-ai-chat-gpt3-key'
+
+// default first message to display in UI (not necessary to define the prompt)
+export const initialMessages: Message[] = [
+  {
+    who: 'bot',
+    message: 'Hi! I’m A friendly AI assistant. Ask me anything!',
+  },
+]
+
+export function Chat() {
+  const [messages, setMessages] = useState<Message[]>(initialMessages)
+  const [input, setInput] = useState('')
+  const [loading, setLoading] = useState(false)
+  const [cookie, setCookie] = useCookies([COOKIE_NAME, COOKIE_KEY])
+
+  const [botName, setBotName] = useState('AI')
+  const [userName, setUserName] = useState('News reporter')
+  const [keyOpenAI, setKeyOpenAI] = useState('sk-XXXX') // default OpenAI key
```

Comment:
I don't see anywhere in the OpenAI docs that they have client-side keys, so this looks wrong and is not even using an environment variable 🤔